test/test_file.cpp

```diff
@@ -408,3 +399,120 @@ TORRENT_TEST(stat_file)
 	TEST_CHECK(ec);
 	TEST_EQUAL(ec, boost::system::errc::no_such_file_or_directory);
 }
+
+// specificaly UNC tests
+#if TORRENT_USE_UNC_PATHS
+
+int maximum_component_length { 0 };
+bool support_hard_links { false };
+
+bool fill_current_directory_caps()
+{
+#ifdef TORRENT_WINDOWS
+	error_code ec;
+	DWORD dw_maximum_component_length;
+	DWORD dw_file_system_flags;
+	if (!GetVolumeInformationA(nullptr, nullptr, 0, nullptr, &dw_maximum_component_length, &dw_file_system_flags, nullptr, 0))
+	{
+		ec.assign(GetLastError(), system_category());
+		std::printf("GetVolumeInformation: [%s] %s\n"
+			, ec.category().name(), ec.message().c_str());
+		return false;
+	}
+	maximum_component_length = int(dw_maximum_component_length);
+	support_hard_links = ((dw_file_system_flags & FILE_SUPPORTS_HARD_LINKS) != 0);
+	return maximum_component_length > 0;
+#else
+	return false;
+#endif
+}
+
+TORRENT_TEST(unc_tests)
+{
+	TEST_EQUAL(canonicalize_path("c:\\a\\..\\b"), "c:\\b");
+	TEST_EQUAL(canonicalize_path("a\\..\\b"), "b");
+	TEST_EQUAL(canonicalize_path("a\\..\\.\\b"), "b");
+	TEST_EQUAL(canonicalize_path("\\.\\a"), "\\a");
+	TEST_EQUAL(canonicalize_path("\\\\bla\\.\\a"), "\\\\bla\\a");
+	TEST_EQUAL(canonicalize_path("c:\\bla\\a"), "c:\\bla\\a");
+
+	error_code ec;
+
+	std::vector<std::string> special_names
+	{
+		"CON", "PRN", "AUX", "NUL",
+		"COM1", "COM2", "COM3", "COM4",
+		"COM5", "COM6", "COM7", "COM8",
+		"COM9", "LPT1", "LPT2", "LPT3",
+		"LPT4", "LPT5", "LPT6", "LPT7",
+		"LPT8", "LPT9"
+	};
+
+	for (std::string special_name : special_names)
+	{
+		TEST_EQUAL(touch_file(special_name, 10), 0);
+		TEST_CHECK(lt::exists(special_name));
+		lt::remove(special_name, ec);
+		TEST_EQUAL(ec, error_code());
+		TEST_CHECK(!lt::exists(special_name));
+	}
+
+	if (fill_current_directory_caps())
```

Comment:
the whole ``true`` block below, does it really only make sense to run that on windows?